### PR TITLE
MM-17555 Finish upgrading react-intl

### PR DESCRIPTION
**Pull request by hmhealey**
This is the final PR to update to the new version of react-intl. Most of it should be nonfunctional stuff like updating snapshots and tests, but there's a few important things I wanted to call out:
1. ~~The biggest change that I didn't expect is that the version of react-bootstrap's `OverlayTrigger` that we use doesn't pass context correctly, so I had to make a custom version of `OverlayTrigger` to make it so that modals and popovers (like the profile popover) could be translated correctly.~~ This change was merged as a separate PR.
2. This is using @cometkim's new versions of `shallowWithIntl` and `mountWithIntl`, but I removed the step of them that unwraps calls to `React.forwardRef` since it wasn't necessary and it caused some problems when we were trying to test components that used `forwardRef` when it wasn't from `injectIntl`
3. ~~Related to the above, `shallowWithIntl` is only needed for testing components that use `injectIntl`, so I've removed it from all tests where it wasn't needed. Similarly, we no longer need to `.dive` into most components when testing them.~~ This change was merged in a separate PR.
4. React-intl now uses `React.forwardRef` to pass refs into `injectIntl` components. This was the big source of issues last time that we attempted this upgrade, but I think I've gone through and either tested or removed everywhere we did this

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-17555

### CODE REVIEW

**Comment by enahum** [comment]
@hmhealey can you push this PR to be tested first before I do the review, it is a very large PR and I would rather review it after having confirmation that nothing is broken.

**Comment by cometkim** [comment]
Not only OverlayTrigger, but also Tooltips have same context issue as well. it completely destroys the context API usage and causes memory leaks in VDOM. Wouldn't it be better to upgrade react-bootstrap first, even if it spends more time?

**Comment by hmhealey** [comment]
@enahum I'll see if I can do that, although we're in the release cycle right now, so it might be difficult. If it can't be reviewed right now, I'll look at breaking up this PR some more.

@cometkim I could implement a custom Tooltip replacement like OverlayTrigger, but what do you mean by it causing memory leaks? I wasn't aware of anything like that

**Comment by cometkim** [comment]
@hmhealey never mind. I suspected that because elements are duplicated in the inspector and never removed. It seems to be a bug in the inspector.

**Comment by hmhealey** [comment]
I'm going to break this up into a few more PRs wherever possible. I'll re-add reviewers once those PRs are in

**Comment by hmhealey** [comment]
@cometkim Have you found any specific instances where the Tooltip isn't passing the context correctly? It seems to be working fine everywhere I've checked

**Comment by hmhealey** [comment]
This is ready for review again now that some of the large, distracting changes have been merged separately.

For dev reviewers, if you filter out the changes to the `.snap` files in GitHub, it'll make your live's a lot easier since there's a lot of renaming going on caused by `react-intl` renaming `InjectIntl` to `injectIntl` as the component's display name

**Comment by jespino** [comment]
@hmhealey test are failing, probably related to the merge done

File: components/create_comment/create_comment.jsx [hunk 1]
```diff
@@ -704,8 +704,8 @@ class CreateComment extends React.PureComponent {
             if (index !== -1) {
                 uploadsInProgress.splice(index, 1);
 
-                if (this.refs.fileUpload && this.refs.fileUpload.getWrappedInstance() && this.refs.fileUpload.getWrappedInstance().getWrappedInstance()) {
-                    this.refs.fileUpload.getWrappedInstance().getWrappedInstance().cancelUpload(id);
+                if (this.refs.fileUpload && this.refs.fileUpload.getWrappedInstance()) {
```
Comment (enahum) [inline components/create_comment/create_comment.jsx]: do you need `getWrappedInstance` even by using `forwardRef` ? that seems strange

**Comment by enahum** [comment]
@hmhealey PR needs updating, @jgilliam17 please try and test the PR soon ;)

File: components/create_comment/create_comment.jsx (hunk 1 above)
Comment (hmhealey) [reply components/create_comment/create_comment.jsx]: The remaining `getWrappedInstance` is actually from `connect`. If that gets updated to use `React.forwardRef`, then we can remove the second `getWrappedInstance`

**Pull request merged**